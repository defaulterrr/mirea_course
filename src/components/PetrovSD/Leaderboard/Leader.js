import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';


const getService = (link) => {
  if (link.includes('twitch')) {
    return 'Twitch';
  }
  if (link.includes('youtu.be') || link.includes('youtube.com')) {
    return 'Youtube';
  }
  return 'Watch';
};

const getTime = (time) => {
  const mins = (time - (time % 60)) / 60;
  const secs = time - mins * 60;
  return `${mins}m ${secs}sec`;
};

const fetchData = async (props, link, stateUpdate) => {
  fetch(link)
    .then((response) => response.json())
    .then((json) => {
      stateUpdate(
        {
          isLoading: false,
          rank: props.rank,
          player: json.data.names.international,
          igt: getTime(props.igt),
          platform: props.platform,
          date: props.date,
          link: props.link,
          loadLink: props.loadLink,
        },
      );
    });
};

const Leader = (props) => {
  const { LoadLink } = props;
  const [state, setState] = useState({
    isLoading: true,
    rank: null,
    player: null,
    igt: null,
    platform: null,
    date: null,
    link: null,
    loadLink: LoadLink,
  });

  useEffect(() => {
    fetchData(props, props.LoadLink, setState);
  }, [props]);

  if (state.isLoading === false) {
    return (
      <tr key={props.key} className="petrov_sd_tr-leader">
        <th className="petrov_sd_th-leader">{state.rank}</th>
        <th className="petrov_sd_th-leader">{state.player}</th>
        <th className="petrov_sd_th-leader">{state.igt}</th>
        <th className="petrov_sd_th-leader">{state.platform}</th>
        <th className="petrov_sd_th-leader">{state.date}</th>
        <th className="petrov_sd_th-leader">
          <a href={state.link} target="_blank" rel="noopener noreferrer">
            {getService(state.link)}
          </a>
        </th>
      </tr>
    );
  }
  return (
    <tr>
      <th>Fetching...</th>
      <th>Fetching...</th>
      <th>Fetching...</th>
      <th>Fetching...</th>
      <th>Fetching...</th>
      <th>Fetching...</th>
    </tr>
  );
};

Leader.propTypes = {
  LoadLink: PropTypes.string.isRequired,
  key: PropTypes.string.isRequired,
};

export default Leader;
