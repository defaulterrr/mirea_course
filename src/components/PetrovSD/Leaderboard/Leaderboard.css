.petrov_sd_th-leader{
    padding: 10px;
}

.petrov_sd_th-tr{
    height: 20px;
}