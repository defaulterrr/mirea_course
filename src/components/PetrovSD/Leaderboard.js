import React, {useState, useEffect} from 'react'
import Leader from './Leaderboard/Leader'
import './Leaderboard/Leaderboard.css'

const fetchData = async (stateUpdate) => {
    
    fetch("https://www.speedrun.com/api/v1/leaderboards/lde3woe6/category/ndx1pm52?top=10")
    .then(response => response.json())
    .then(json => {
        stateUpdate({
            leaders: json.data.runs,
            isLoading: false
        })
        console.log(json)    
    }
    )
}

const Leaderboard = (props) => {
    const [ state,setState ] = useState({
        leaders: props.leaders || null,
        isLoading: props.isLoading || true
    })

    useEffect(() => {
        fetchData(setState)
    },[])

    return (
        <div className="Leaderboard">
            <p>Dark Souls Remastered Leaderboards (Any%)</p>
            <table className='leaderboardTable'>
                <th>Rank</th>
                <th>Player</th>
                <th>IGT</th>
                <th>Platform</th>
                <th>Date</th>
                <th>Link</th>
            {state.isLoading === false ? state.leaders.map((chunk)=> {
                return (
                    <Leader rank={chunk.place} igt={chunk.run.times.ingame_t} platform="PC" date={chunk.run.date} loadLink={chunk.run.players[0].uri} link={chunk.run.videos.links[0].uri}/>
                )
            }) : null}
            </table>
        </div>
    )
}

export default Leaderboard