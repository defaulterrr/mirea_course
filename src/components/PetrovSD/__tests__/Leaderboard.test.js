import React from 'react'
import { create } from 'react-test-renderer'
import Leaderboard from '../Leaderboard'

const testData = {"data":{"weblink":"https://www.speedrun.com/darksoulsremastered#Any","game":"lde3woe6","category":"ndx1pm52","level":null,"platform":null,"region":null,"emulators":null,"video-only":false,"timing":"ingame","values":{},"runs":[{"place":1,"run":{"id":"z5j327em","weblink":"https://www.speedrun.com/darksoulsremastered/run/z5j327em","game":"lde3woe6","level":null,"category":"ndx1pm52","videos":{"links":[{"uri":"https://www.twitch.tv/videos/610640194"}]},"comment":"I have done force quit by mistake in NG+","status":{"status":"verified","examiner":"dx36pd2x","verify-date":"2020-05-07T11:35:13Z"},"players":[{"rel":"user","id":"zxzw45nx","uri":"https://www.speedrun.com/api/v1/users/zxzw45nx"}],"date":"2020-05-04","submitted":"2020-05-04T04:27:55Z","times":{"primary":"PT30M11S","primary_t":1811,"realtime":null,"realtime_t":0,"realtime_noloads":null,"realtime_noloads_t":0,"ingame":"PT30M11S","ingame_t":1811},"system":{"platform":"8gej2n93","emulated":false,"region":null},"splits":null,"values":{}}},{"place":2,"run":{"id":"y4w95wnm","weblink":"https://www.speedrun.com/darksoulsremastered/run/y4w95wnm","game":"lde3woe6","level":null,"category":"ndx1pm52","videos":{"links":[{"uri":"https://youtu.be/TdAJiHvQNoU"}]},"comment":null,"status":{"status":"verified","examiner":"o86ddg0j","verify-date":"2020-03-26T06:14:25Z"},"players":[{"rel":"user","id":"y8dywvm8","uri":"https://www.speedrun.com/api/v1/users/y8dywvm8"}],"date":"2020-03-23","submitted":"2020-03-23T17:09:18Z","times":{"primary":"PT31M4S","primary_t":1864,"realtime":null,"realtime_t":0,"realtime_noloads":null,"realtime_noloads_t":0,"ingame":"PT31M4S","ingame_t":1864},"system":{"platform":"8gej2n93","emulated":false,"region":null},"splits":null,"values":{}}},{"place":3,"run":{"id":"y6l1ko0y","weblink":"https://www.speedrun.com/darksoulsremastered/run/y6l1ko0y","game":"lde3woe6","level":null,"category":"ndx1pm52","videos":{"links":[{"uri":"https://www.twitch.tv/videos/566609489"}]},"comment":"i still can\u0027t believe it was a legit run :Wowega:","status":{"status":"verified","examiner":"o86ddg0j","verify-date":"2020-03-15T14:02:30Z"},"players":[{"rel":"user","id":"dx36pd2x","uri":"https://www.speedrun.com/api/v1/users/dx36pd2x"}],"date":"2020-03-13","submitted":"2020-03-13T18:32:07Z","times":{"primary":"PT31M24S","primary_t":1884,"realtime":null,"realtime_t":0,"realtime_noloads":null,"realtime_noloads_t":0,"ingame":"PT31M24S","ingame_t":1884},"system":{"platform":"8gej2n93","emulated":false,"region":null},"splits":null,"values":{}}},{"place":4,"run":{"id":"y41435ky","weblink":"https://www.speedrun.com/darksoulsremastered/run/y41435ky","game":"lde3woe6","level":null,"category":"ndx1pm52","videos":{"links":[{"uri":"https://www.youtube.com/watch?v=rbtq5pQ5Aqc"}]},"comment":"(no audio, had it off in OBS) Trash run, lots of timesaves, really happy i got sub 32 though. :)","status":{"status":"verified","examiner":"o86ddg0j","verify-date":"2020-02-23T04:29:35Z"},"players":[{"rel":"user","id":"8dw30q5j","uri":"https://www.speedrun.com/api/v1/users/8dw30q5j"}],"date":"2020-02-22","submitted":"2020-02-22T01:28:33Z","times":{"primary":"PT31M57S","primary_t":1917,"realtime":null,"realtime_t":0,"realtime_noloads":null,"realtime_noloads_t":0,"ingame":"PT31M57S","ingame_t":1917},"system":{"platform":"8gej2n93","emulated":false,"region":null},"splits":null,"values":{}}},{"place":5,"run":{"id":"m31q956z","weblink":"https://www.speedrun.com/darksoulsremastered/run/m31q956z","game":"lde3woe6","level":null,"category":"ndx1pm52","videos":{"links":[{"uri":"https://youtu.be/FHk1hZnqbsA"}]},"comment":"garbage run don\u0027t watch","status":{"status":"verified","examiner":"o86ddg0j","verify-date":"2019-11-12T00:09:29Z"},"players":[{"rel":"user","id":"qxk5q27x","uri":"https://www.speedrun.com/api/v1/users/qxk5q27x"}],"date":"2019-11-11","submitted":"2019-11-12T00:00:38Z","times":{"primary":"PT32M18S","primary_t":1938,"realtime":null,"realtime_t":0,"realtime_noloads":null,"realtime_noloads_t":0,"ingame":"PT32M18S","ingame_t":1938},"system":{"platform":"8gej2n93","emulated":false,"region":null},"splits":null,"values":{}}},{"place":6,"run":{"id":"z5jje65m","weblink":"https://www.speedrun.com/darksoulsremastered/run/z5jje65m","game":"lde3woe6","level":null,"category":"ndx1pm52","videos":{"links":[{"uri":"https://www.youtube.com/watch?v=e31GIOaJbxY"}]},"comment":"\u7d0d\u5f97\u3067\u304d\u306a\u3044\u3002I\u0027m not satisfied this running.","status":{"status":"verified","examiner":"dx36pd2x","verify-date":"2020-05-01T14:37:07Z"},"players":[{"rel":"user","id":"8l0pder8","uri":"https://www.speedrun.com/api/v1/users/8l0pder8"}],"date":"2020-04-27","submitted":"2020-04-27T12:15:58Z","times":{"primary":"PT32M23S","primary_t":1943,"realtime":null,"realtime_t":0,"realtime_noloads":null,"realtime_noloads_t":0,"ingame":"PT32M23S","ingame_t":1943},"system":{"platform":"nzelkr6q","emulated":false,"region":null},"splits":null,"values":{}}},{"place":7,"run":{"id":"mreeqkdy","weblink":"https://www.speedrun.com/darksoulsremastered/run/mreeqkdy","game":"lde3woe6","level":null,"category":"ndx1pm52","videos":{"links":[{"uri":"https://youtu.be/7iqKVLCEHjA"}]},"comment":"Pretty bad.","status":{"status":"verified","examiner":"dx36pd2x","verify-date":"2019-06-24T17:46:28Z"},"players":[{"rel":"user","id":"o86ddg0j","uri":"https://www.speedrun.com/api/v1/users/o86ddg0j"}],"date":"2019-06-24","submitted":"2019-06-24T17:11:53Z","times":{"primary":"PT32M25S","primary_t":1945,"realtime":null,"realtime_t":0,"realtime_noloads":null,"realtime_noloads_t":0,"ingame":"PT32M25S","ingame_t":1945},"system":{"platform":"8gej2n93","emulated":false,"region":null},"splits":null,"values":{}}},{"place":8,"run":{"id":"mk923lvz","weblink":"https://www.speedrun.com/darksoulsremastered/run/mk923lvz","game":"lde3woe6","level":null,"category":"ndx1pm52","videos":{"links":[{"uri":"https://youtu.be/bac8YDHFvFY"}]},"comment":"Crazy run, but definitely improvements to be made","status":{"status":"verified","examiner":"zx7qwnr8","verify-date":"2018-07-10T16:15:04Z"},"players":[{"rel":"user","id":"o86vmv3j","uri":"https://www.speedrun.com/api/v1/users/o86vmv3j"}],"date":"2018-07-08","submitted":"2018-07-10T10:37:05Z","times":{"primary":"PT32M54S","primary_t":1974,"realtime":"PT35M51S","realtime_t":2151,"realtime_noloads":null,"realtime_noloads_t":0,"ingame":"PT32M54S","ingame_t":1974},"system":{"platform":"8gej2n93","emulated":false,"region":null},"splits":null,"values":{}}},{"place":9,"run":{"id":"zg7317ez","weblink":"https://www.speedrun.com/darksoulsremastered/run/zg7317ez","game":"lde3woe6","level":null,"category":"ndx1pm52","videos":{"links":[{"uri":"https://www.youtube.com/watch?v=q9RC3xuRnw0"}]},"comment":null,"status":{"status":"verified","examiner":"zx7qwnr8","verify-date":"2018-06-04T23:18:43Z"},"players":[{"rel":"user","id":"zx7qwnr8","uri":"https://www.speedrun.com/api/v1/users/zx7qwnr8"}],"date":"2018-06-04","submitted":"2018-06-04T23:18:43Z","times":{"primary":"PT32M56S","primary_t":1976,"realtime":null,"realtime_t":0,"realtime_noloads":null,"realtime_noloads_t":0,"ingame":"PT32M56S","ingame_t":1976},"system":{"platform":"8gej2n93","emulated":false,"region":null},"splits":null,"values":{}}},{"place":10,"run":{"id":"mr8wn67y","weblink":"https://www.speedrun.com/darksoulsremastered/run/mr8wn67y","game":"lde3woe6","level":null,"category":"ndx1pm52","videos":{"links":[{"uri":"https://www.twitch.tv/videos/399493884"}]},"comment":"MOD edit: incorrect split timing, IGT stops on credits. Time moved from 32:54 to 32:57.\r\n\r\nplayer comment: sorry for incorrect split timing. thank you moderator for telling me this :)","status":{"status":"verified","examiner":"dx36pd2x","verify-date":"2019-03-27T13:21:10Z"},"players":[{"rel":"user","id":"y8dkwzlj","uri":"https://www.speedrun.com/api/v1/users/y8dkwzlj"}],"date":"2019-03-23","submitted":"2019-03-22T20:04:42Z","times":{"primary":"PT32M57S","primary_t":1977,"realtime":null,"realtime_t":0,"realtime_noloads":null,"realtime_noloads_t":0,"ingame":"PT32M57S","ingame_t":1977},"system":{"platform":"8gej2n93","emulated":false,"region":null},"splits":null,"values":{}}}],"links":[{"rel":"game","uri":"https://www.speedrun.com/api/v1/games/lde3woe6"},{"rel":"category","uri":"https://www.speedrun.com/api/v1/categories/ndx1pm52"}]}}

describe("[RENDER] Calculator â€“ Snapshot test", ()=> {
    test("Matches the snapshot", ()=>{
        const leaderboard = create(<Leaderboard leaders={testData.data.runs} isLoading={false}/>)
        expect(leaderboard.toJSON()).toMatchSnapshot()
    })
})