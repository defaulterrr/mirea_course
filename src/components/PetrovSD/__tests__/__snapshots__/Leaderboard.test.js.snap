// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] Calculator â€“ Snapshot test Matches the snapshot 1`] = `
<div
  className="Leaderboard"
>
  <p>
    Dark Souls Remastered Leaderboards (Any%)
  </p>
  <table
    className="leaderboardTable"
  >
    <th>
      Rank
    </th>
    <th>
      Player
    </th>
    <th>
      IGT
    </th>
    <th>
      Platform
    </th>
    <th>
      Date
    </th>
    <th>
      Link
    </th>
  </table>
</div>
`;
